framework/src/parser/Parser.C

```diff
@@ -69,58 +69,157 @@ Parser::Parser(MooseApp & app, ActionWarehouse & action_wh)
 Parser::~Parser() {}
 
 bool
-Parser::isSectionActive(const std::string & s,
-                        const std::map<std::string, BlockLists> & block_lists) const
+isSectionActive(std::string path, hit::Node * root)
 {
-  /**
-   * The default return value is false to support the case where the active list is explicitly set
-   * to empty (active = ''). It's also possible to have an empty inactive list (default) in which
-   * case we don't change this value throughout this routine.
-   */
-  bool retValue = false;
-  auto found = s.find_last_of('/');
+  hit::Node * n = root->find(path);
+  while (n != nullptr)
+  {
+    hit::Node * section = n->parent();
+    if (section)
+    {
+      auto an = section->find("active");
+      auto ian = section->find("inactive");
```

Comment:
Not a huge fan of these names. They are bad. If you start in the code below, these names mean nothing.